app/models/manageiq/providers/ansible_tower/shared/automation_manager/machine_credential.rb

```diff
@@ -0,0 +1,55 @@
+module ManageIQ::Providers::AnsibleTower::Shared::AutomationManager::MachineCredential
+  extend ActiveSupport::Concern
+
+  COMMON_ATTRIBUTES = {
+    :userid => {
+      :label     => N_('Username'),
+      :help_text => N_('Username for this credential')
+    },
+    :password => {
+      :type      => :password,
+      :label     => N_('Password'),
+      :help_text => N_('Password for this credential')
+    }
+  }.freeze
+
+  EXTRA_ATTRIBUTES = {
+    :ssh_key_data => {
+      :type       => :password,
+      :label      => N_('Private key'),
+      :help_text  => N_('RSA or DSA private key to be used instead of password')
+    },
+    :ssh_key_unlock => {
+      :type       => :password,
+      :label      => N_('Private key passphrase'),
+      :help_text  => N_('Passphrase to unlock SSH private key if encrypted'),
+      :max_length => 1024
+    },
+    :become_method => {
+      :type       => :choice,
+      :label      => N_('Privilege Escalation'),
+      :help_text  => N_('Privilege escalation method'),
+      :choices    => %w(sudo su pbrun pfexec).insert(0, '')
```

Comment:
@jameswnl What is the purpose of the `insert(0, '')`.  Is blank a valid choice, or is this for UI purposes?  If the latter, then that's a UI concern, and should not be here in the model at all.